crates/api/src/local_user/ban_person.rs

```diff
@@ -61,15 +62,28 @@ pub async fn ban_from_site(
   // Mod tables
   let form = ModBanForm {
     mod_person_id: local_user_view.person.id,
-    other_person_id: data.person_id,
+    other_person_id: person.id,
     reason: data.reason.clone(),
     banned: Some(data.ban),
     expires,
   };
 
   ModBan::create(&mut context.pool(), &form).await?;
 
-  let person_view = PersonView::read(&mut context.pool(), data.person_id).await?;
+  let person_view = PersonView::read(&mut context.pool(), person.id).await?;
+
+  if !person.local {
```

Comment:
Wondering if I should move this check into the function itself. Doesn't matter to me.

Reply:
The function is called `nonlocal_user`, so it makes sense that it performs the check internally.